### PR TITLE
Topic/ak/rdpeudp

### PR DESCRIPTION
**Pull request by anthonykasza**
This includes more robust parsing of PDUs for RDPEUDP. The analyzer now properly tracks negotiation of RDPEUDP2. My apologies for submitting this work as multiple PRs.

### CODE REVIEW

File: src/analyzer/protocol/rdp/rdpeudp-analyzer.pac [hunk 1]
```diff
@@ -19,12 +23,12 @@ refine connection RDPEUDP_Conn += {
 		return state_;
 		%}
 
-	function is_version2(): bool
+	function is_rdpeudp2(): bool
 		%{
-		return (orig_rdpeudp2_ && resp_rdpeudp2_);
+		return ((orig_synex_flags_ & resp_synex_flags_) >= RDPUDP_PROTOCOL_VERSION_3);
```
Comment (jsiwek) [inline src/analyzer/protocol/rdp/rdpeudp-analyzer.pac:28]: Maybe equality comparisons would be clearer?

```suggestion
		return orig_synex_flags_  == RDPUDP_PROTOCOL_VERSION_3 && resp_synex_flags_ == RDPUDP_PROTOCOL_VERSION_3;
```

File: src/analyzer/protocol/rdp/rdpeudp-protocol.pac [hunk 2]
```diff
@@ -26,11 +26,38 @@ type RDPEUDP_PDU(is_orig: bool) = record {
...
-	proc_rdpeudp_syn: bool = $context.connection.proc_rdpeudp_syn(is_orig, fec_header.uFlags, fec_header.snSourceAck);
+	proc_rdpeudp_syn: bool = $context.connection.proc_rdpeudp_syn(is_orig, fec_header.uFlags,
+							fec_header.snSourceAck, has_synex_payload.uUdpVer);
+};
+
+# The tech specs refer to this as both RDPUDP_SYNEX_PAYLOAD and RDPUDP_SYNDATAEX_PAYLOAD
+type RDPUDP_SYNEX_PAYLOAD = record {
+	uSynExFlags:	uint16;
+	uUdpVer:	uint16;
+	cookieHash:	case ((uUdpVer & RDPUDP_PROTOCOL_VERSION_3) > 0) of {
```
Comment (jsiwek) [inline src/analyzer/protocol/rdp/rdpeudp-protocol.pac:46]: Equality comparison may be clearer.

```suggestion
	cookieHash:	case (uUdpVer == RDPUDP_PROTOCOL_VERSION_3) of {
```

File: testing/btest/scripts/base/protocols/rdp/rdpeudp2-handshake-success.zeek [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# @TEST-EXEC: zeek -r $TRACES/rdp/rdpeudp2-handshake-success.pcap %INPUT >out
+# @TEST-EXEC: btest-diff conn.log
+# @TEST-EXEC: btest-diff out
```
Comment (jsiwek) [inline testing/btest/scripts/base/protocols/rdp/rdpeudp2-handshake-success.zeek:3]: * The baseline for this wasn't included in the PR, can generate via `btest -U scripts/base/protocols/rdp/rdpeudp2-handshake-success.zeek` and add/commit the new files in `Baseline/`.  (Also CI shows the previous test for `rdpeudp-handshake-success.zeek` should update its baseline)
* I noticed this one's output shows only the `rdpeudp_syn` event, but nothing else, were you expecting that or should it really be triggering the synac/established/data events, too?

Comment (anthonykasza) [reply testing/btest/scripts/base/protocols/rdp/rdpeudp2-handshake-success.zeek:3]: I've updated the btest Baselines. The events are now being raised by rdpeudp2-handshake-success.zeek.

**Pull request merged**